min_versions.py

```diff
@@ -170,6 +179,39 @@ def find_pkg_mismatches(pkg_name, pkg_version, fnames):
     return pkg_mismatches
 
 
+def test_pkg_mismatch_regex():
+    """
+    Validation function for the package mismatch regex
+    """
+    pkgs = {
+        "numpy": "0.0",
+        "scipy": "0.0",
+        "python": "2.7",
+        "python-version": "2.7",
+        "numba": "0.0",
+    }
+
+    lines = [
+        "Programming Language :: Python :: 3.8",
+        "STUMPY supports Python 3.8",
+        "python-version: ['3.8']",
+        'requires-python = ">=3.8"',
+        "numba>=0.55.2",
+    ]
+
+    matches_found = []
+    for pkg_name, pkg_version in pkgs.items():
+
+        for line in lines:
+            matches = match_str_version(line, pkg_name)
+
+            if matches:
+                matches_found.append(matches.group(0))
+
+    if len(matches_found) != len(lines):
+        raise ValueError("Edge case not matched by regex")
```

Comment:
I don't think that this is a good inequality check as it assumes that each of the matches found correspond to each line found in lines. If an error is made in our code/regex then it might be possible, for example, that a single line could produce multiple matches. 

Perhaps it would be "better" if we swapped the order of the `for` loops so that we iterate over lines first and then packages and `break` if a match is found but raise an error if no match is found and then provide a specific error message that identifies precisely which `line` is not being discovered by the regex.

Thoughts?